omegatrade-app/frontend/src/app/components/stock-dashboard/stock-dashboard.component.ts

```diff
@@ -0,0 +1,218 @@
+
+  selectedCompany: string = "";
+  lastUpdatedTime;
+  stock: StockChart;
+  companies: any;
+  loader: boolean = false;
+  subscription: Subscription;
+  timerIds = [];
+  
+  constructor(private snackBarService: SnackBarService, private router: ActivatedRoute, private _snackBar: MatSnackBar, private restService: RestService) {
+  }
+
+  /**
+   *  Function to Initiate component.
+   *  Initiating company lists
+   */
+  ngOnInit(): void {
+    this.selectedCompany = this.router.snapshot.queryParamMap.get('companyId');
+    this.getCompanies();
+  }
+
+  /**
+   *  Function to get all companies.
+   *  @returns  {null}
+   */
+  getCompanies() {
+    this.restService.getData('companies/list')
+      .pipe(take(1))
+      .subscribe(
+        response => {
+          if (response && response.success) {
+            this.companies = response.data;
+            if (this.companies && this.companies.length > 0) {
+              if(!this.selectedCompany || this.selectedCompany === "")
+                this.selectedCompany = this.companies[0].companyId;
+              this.getStockData();
+            }
+          }
+        },
+        error => {
+          if (error && error.error && error.error.message) {
+            this.snackBarService.openSnackBar(error.error.message, '');
+          }
+          this.clearAllTimeOuts();
+          this.loader = false;
+        });
+  }
+
+  /**
+   *  Function to get all stockDatas for selected company.
+   * 
+   *  @returns  {null}
+   */
+  getStockData() {
+    if (this.selectedCompany) {
+      this.loader = true;
+      this.restService.getData(`companies/dashboard/${this.selectedCompany}`)
+        .pipe(take(1))
+        .subscribe(
+          response => {
+            if (response && response.success) {
+              const stocks = response.data.stocks;
+              const company = response.data.company;
+              if (stocks && stocks.length > 0) {
+                this.parseStockDatas(stocks,company);
+              } else if (company && company.status === 'PROCESSING') {
+                /**
+                 * Retrying getStockData in the case of empty stocks at that current time. 
+                 * But it may have data since the staus is in PROCESSING,
+                 * so fetching the datas of running simulation in certain interval.
+                 * 
+                 */
+                const id = setTimeout(() => {
+                  this.getStockData();
+                }, 5000);
+                this.timerIds.push(id);
+              }
+              this.loader = false;
+            }
+          },
+          error => {
+            if (error && error.error && error.error.message) {
+              this.snackBarService.openSnackBar(error.error.message, '');
+            }
+            this.clearAllTimeOuts();
+            this.loader = false;
+          });
+    }
+  }
+  /**
+   * Function to parse the stocks and form as per the chart data format
```

Comment:
I don't think the name (and comment) of this function reflects very well what it does. This function will not only parse the stock data, but also update the `lastUpdatedTime` and redraw the chart.

Reply:
the above issue has been resolved.